src/Sparrow/Json/BlittableJsonReaderArray.cs

```diff
@@ -71,52 +75,24 @@ public void BlittableValidation()
             _parent?.BlittableValidation();
         }
 
-        //Todo Fixing the clone implementation to support this situation or throw clear error
+        public BlittableJsonReaderArray Clone(BlittableJsonDocumentBuilder.UsageMode usageMode = BlittableJsonDocumentBuilder.UsageMode.None)
+        {
+            return Clone(_context, usageMode);
+        }
+        
         public BlittableJsonReaderArray Clone(JsonOperationContext context, BlittableJsonDocumentBuilder.UsageMode usageMode = BlittableJsonDocumentBuilder.UsageMode.None)
         {
             AssertContextNotDisposed();
-            using (var builder = new ManualBlittableJsonDocumentBuilder<UnmanagedWriteBuffer>(context))
-            {
-                builder.Reset(usageMode);
-                builder.StartArrayDocument();
-                builder.StartWriteArray();
-                using (var itr = new BlittableJsonArrayEnumerator(this))
-                {
-                    while (itr.MoveNext())
-                    {
-                        switch (itr.Current)
-                        {
-                            case BlittableJsonReaderObject item:
-                                var clone = item.CloneOnTheSameContext();
-                                builder.WriteEmbeddedBlittableDocument(clone.BasePointer, clone.Size);
-                                break;
-
-                            case LazyStringValue item:
-                                builder.WriteValue(item);
-                                break;
-
-                            case long item:
-                                builder.WriteValue(item);
-                                break;
-
-                            case LazyNumberValue item:
-                                builder.WriteValue(item);
-                                break;
-
-                            case LazyCompressedStringValue item:
-                                builder.WriteValue(item);
-                                break;
-
-                            default:
-                                throw new InvalidDataException($"Actual value type is {itr.Current.GetType()}. Should be known serialized type and should not happen. ");
-                        }
-                    }
-                }
-                builder.WriteArrayEnd();
-                builder.FinalizeDocument();
 
-                return builder.CreateArrayReader();
-            }
+            var arrayHolder = IsRoot() && Modifications == null
+                ? _parent.Clone(context) 
+                : context.ReadObject(new DynamicJsonValue { [RootArrayHolderPropertyName] = this }, "array holder", usageMode);
+            
+            if(arrayHolder.TryGet(RootArrayHolderPropertyNameSegment, out BlittableJsonReaderArray array) == false)
+                throw new InvalidOperationException("Couldn't find array");
+                
+            array.ArrayIsRoot();
```

Comment:
When checking where else we call it I found the following code:

https://github.com/ravendb/ravendb/blob/1093bac5556323dfa3f8c4947529d8bff1780b99/src/Sparrow/Json/BlittableJsonDocumentBuilder.cs#L773-L783

Can we use `RootArrayHolderPropertyNameSegment` there instead of `"_"`?